File: backend/api/src/common/errors/custom-exceptions.ts
import { HttpException, HttpStatus } from '@nestjs/common';

/**
 * ValidationException
 * Custom exception for input validation errors.
 */
export class ValidationException extends HttpException {
    constructor(message: string) {
        super(message, HttpStatus.BAD_REQUEST);
    }
}

/**
 * DatabaseException
 * Custom exception for database-related errors.
 */
export class DatabaseException extends HttpException {
    constructor(message: string) {
        super(message, HttpStatus.INTERNAL_SERVER_ERROR);
    }
}

File: backend/ai/app/errors/custom_exceptions.py
from fastapi import HTTPException

class ValidationException(HTTPException):
    """
    ValidationException
    Custom exception for input validation errors.
    """
    def __init__(self, detail: str):
        super().__init__(status_code=400, detail=detail)


class ModelProcessingException(HTTPException):
    """
    ModelProcessingException
    Custom exception for errors during model processing.
    """
    def __init__(self, detail: str):
        super().__init__(status_code=500, detail=detail)
